### PR TITLE
feat(labeler): Cloudflare Access operator authentication (W9.1)

### PR DESCRIPTION
**Pull request by ascorbic**
## What does this PR do?

Adds the labeler operator console's **Cloudflare Access authentication module** — plan item W9.1, spec §12. `verifyAccessRequest` establishes an `OperatorIdentity` from the `Cf-Access-Jwt-Assertion` header before any console route (W9.2+) will permit moderation actions.

Design:

- **Identity comes only from a cryptographically verified assertion.** The module reads exactly one header — `Cf-Access-Jwt-Assertion` — and verifies it with jose against Cloudflare Access's remote JWKS (`${teamDomain}/cdn-cgi/access/certs`). Plaintext identity headers like `Cf-Access-Authenticated-User-Email` are attacker-spoofable and are never read; a test pins that a request carrying only the plaintext header fails with `missing-token`.
- **Exact issuer/audience pinning, zero clock tolerance.** `issuer` = team domain, `audience` = the Access application AUD tag; `exp`/`nbf` enforced with jose's default zero tolerance. `algorithms` is pinned to `["RS256"]` and `exp` is a required claim — both defense-in-depth (alg-confusion and `alg:none` are already structurally rejected by jose's JWKS key selector, and Access always sets `exp`), so a future resolver change fails hard instead of quietly widening acceptance.
- **Human vs service-token identities.** A `common_name` claim yields a `service` identity, `email` a `human` one; neither yields an identity without a non-empty `sub`. Roles (`admin`/`reviewer`) are mapped by exact `Set` membership of the principal plus signed `groups` claims against config allowlists — empty allowlists fail closed, and `hasRole` gives admins reviewer capability by inheritance.
- **JWKS resolver cached on `globalThis`** (`Symbol.for`, keyed by team domain) so Vite SSR chunk duplication and isolate reuse share one resolver; jose handles kid rotation and cooldown internally.
- **`parseAccessAuthConfig`** validates the deployment config up front: HTTPS team domain, non-empty audience, non-empty string allowlists.

Error surface is two static-message `AccessAuthError` reasons (`missing-token` / `invalid-token`) — token contents are never interpolated into messages.

Two independent opus adversarial passes traced the full jose verification path (key selection, key-type checks, claims validation) from installed source: no authorization path exists without a verified signature, and role claims cannot be self-escalated. Their two recommendations (RS256 pin + alg-confusion/`alg:none` regression tests) are applied here.

No routes are added yet — W9.2 (mutation protections) and W9.3 (console shell) consume this module.

Targets `feat/plugin-registry-labelling-service`. Related to #1909 and RFC #694.

## Type of change

- [ ] Bug fix
- [x] Feature (requires [maintainer-approved Discussion](https://github.com/emdash-cms/emdash/discussions/categories/ideas))
- [ ] Refactor (no behavior change)
- [ ] Translation
- [ ] Documentation
- [ ] Performance improvement
- [x] Tests
- [ ] Chore (dependencies, CI, tooling)

## Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/emdash-cms/emdash/blob/main/CONTRIBUTING.md)
- [x] `pnpm typecheck` passes
- [x] `pnpm lint` passes
- [x] `pnpm test` passes (or targeted tests for my change)
- [x] `pnpm format` has been run
- [x] I have added/updated tests for my changes (if applicable)
- [ ] User-visible strings in the admin UI are [wrapped for translation](https://github.com/emdash-cms/emdash/blob/main/CONTRIBUTING.md#internationalization-i18n) (if applicable). Do not include `messages.po` changes except in translation PRs — a workflow extracts catalogs on merge to `main`.
- [x] I have added a [changeset](https://github.com/emdash-cms/emdash/blob/main/CONTRIBUTING.md#changesets) (if this PR changes a published package)
- [x] New features link to an approved Discussion: RFC #694

Admin i18n and changeset are n/a — this touches only `apps/labeler`, a private, unpublished Worker app (no admin UI yet, not on npm).

## AI-generated code disclosure

- [x] This PR includes AI-generated code — model/tool: Claude Code with Claude Fable 5 (design, coordination), Opus 4.8 subagents (security implementation review and hardening), and a Sonnet subagent (implementation)

## Screenshots / test output

- `pnpm --filter @emdash-cms/labeler test`: 259 pass (17 files) — new suite covers verified accept paths (human, service token, group-derived roles), and rejections for: missing header, spoofed plaintext email header only, garbage token, wrong signing key, wrong issuer, wrong audience, expired, not-yet-valid, missing sub, neither email nor common_name, malformed groups, ES256-signed token, HS256 token whose signature the resolver's key *would* verify (isolates the RS256 pin), unsigned `alg:none` token, and a token omitting `exp`
- Typecheck clean; `pnpm lint:json` diagnostics: 0
- Error messages verified to never contain token material

https://claude.ai/code/session_014rikrGMh25h1rJczUQyTMM

---

### Try this PR

**[Open a fresh playground →](https://feat-labeler-access-auth-emdash-playground.emdash-cms.workers.dev)**

A full working EmDash site, deployed from this branch. Each visit gets its own session-scoped sandbox: no login needed and no shared state. Try the admin, edit content, hit the public site.

<sub>Tracks `feat/labeler-access-auth`. Updated automatically when the playground redeploys.</sub>

### CODE REVIEW

File: apps/labeler/src/access-auth.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	const teamDomain = value.teamDomain;
+	if (typeof teamDomain !== "string" || teamDomain.length === 0)
+		throw new TypeError("Access auth config teamDomain must be a non-empty string");
+	let teamDomainUrl: URL;
+	try {
+		teamDomainUrl = new URL(teamDomain);
+	} catch {
+		throw new TypeError("Access auth config teamDomain must be an HTTPS URL");
+	}
+	if (teamDomainUrl.protocol !== "https:")
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply apps/labeler/src/access-auth.ts]: Applied in 0dc6c3a4 — `parseAccessAuthConfig` now rejects paths, credentials, query, and hash, and normalizes the stored value to `URL.origin` (Access emits `iss` as the bare origin, so a trailing-slash config previously failed every token silently; normalizing keeps the issuer pin byte-exact while accepting both forms). JWKS URL is now built relative to the origin as suggested. Covered by new rejection tests plus a trailing-slash normalization test.

~ 🤖 Fable

File: apps/labeler/src/access-auth.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	const roles: OperatorRole[] = [];
+	if (config.admins.some((admin) => principals.has(admin))) roles.push("admin");
+	if (config.reviewers.some((reviewer) => principals.has(reviewer))) roles.push("reviewer");
+
+	return { ...identity, roles };
+}
+
+function groupPrincipals(groups: unknown): readonly string[] {
+	if (!Array.isArray(groups)) return [];
+	return groups.filter((entry): entry is string => typeof entry === "string");
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply apps/labeler/src/access-auth.ts]: Applied in 0dc6c3a4. Also added a test proving the filter is load-bearing: a token with an empty-string group entry against a (deliberately odd) config allowlist containing "" would otherwise grant admin.

~ 🤖 Fable
